src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -833,6 +835,31 @@ def load_module(name, value):
 
         init_dict = {k: v for k, v in init_dict.items() if load_module(k, v)}
 
+        for key in init_dict.keys():
+            if key not in passed_class_obj:
+                continue
+
+            class_obj = passed_class_obj[key]
+            _expected_class_types = []
+            for expected_type in expected_types[key]:
+                if isinstance(expected_type, enum.EnumMeta):
+                    _expected_class_types.extend(expected_type.__members__.keys())
+                else:
+                    _expected_class_types.append(expected_type.__name__)
+
+            _is_valid_type = class_obj.__class__.__name__ in _expected_class_types
+            if isinstance(class_obj, SchedulerMixin) and not _is_valid_type:
+                _requires_flow_match = any("FlowMatch" in class_type for class_type in _expected_class_types)
+                _is_flow_match = "FlowMatch" in class_obj.__class__.__name__
+                if _requires_flow_match and not _is_flow_match:
+                    logger.warning(f"Expected FlowMatch scheduler, got {class_obj.__class__.__name__}.")
```

Comment:
Probably okay to raise an error here because `scheduler.scale_noise` would raise an error in the flow matching pipelines if a non-FlowMatch scheduler is used.

Reply:
I really don't want to raise any error here because type hint was not something enforced in this library and it is hard even for us to tell which schedulers can be used/cannot.

Reply:
e.g kandinsky if memory serves I think ddim may also works with some of the pipelines, and the compatibility may change